index.js

```diff
@@ -47,26 +58,30 @@ EventResource.prototype.handle = function (ctx, next) {
         }
     }
   };
-
-  if (ctx.method === "POST" && this.events.post) {
-    this.events.post.run(ctx, domain, function(err) {
-      ctx.done(err, result);
-    });
-  } else if (ctx.method === "GET" && this.events.get) {
-    this.events.get.run(ctx, domain, function(err) {
-      ctx.done(err, result);
-    });
-  } else if (ctx.method === "DELETE" && this.events.delete) {
-    this.events.delete.run(ctx, domain, function(err) {
-      ctx.done(err, result);
-    });
-  } else if (ctx.method === "PUT" && this.events.put) {
-    this.events.put.run(ctx, domain, function(err) {
-      ctx.done(err, result);
-    });
-  } else {
-    next();
-  }
-
   
-};
\ No newline at end of file
+  this.beforeRequest(ctx, domain, function(){
+    if (ctx.method === "POST" && this.events.post) {
+      this.events.post.run(ctx, domain, function (err) {
+        ctx.done(err, result);
+      });
+    } else if (ctx.method === "GET" && this.events.get) {
+      this.events.get.run(ctx, domain, function (err) {
+        ctx.done(err, result);
+      });
+    } else if (ctx.method === "DELETE" && this.events.delete) {
+      this.events.delete.run(ctx, domain, function (err) {
+        ctx.done(err, result);
+      });
+    } else if (ctx.method === "PUT" && this.events.put) {
+      this.events.put.run(ctx, domain, function (err) {
+        ctx.done(err, result);
+      });
+    } else if (ctx.method === "HEAD" && this.events.head) {
+      this.events.head.run(ctx, domain, function (err) {
+        ctx.done();
```

Comment:
@emilkoto you need to apply the parameters properly here `ctx.done(err, result)` or whatever you might need here.

Reply:
@emilkoto I think it would be `ctx.done(err, '');`